File: android/entities/src/main/java/com/tiagohs/entities/main_topics/AwardMainTopic.kt
package com.tiagohs.entities.main_topics

import com.google.gson.annotations.SerializedName
import com.tiagohs.entities.awards.NomineeResult
import com.tiagohs.entities.contents.ContentNominee
import com.tiagohs.entities.awards.Social
import com.tiagohs.entities.contents.Content
import com.tiagohs.entities.image.Image
import java.io.Serializable

data class AwardMainTopic(
    @SerializedName("id")
    val id: Int,

    @SerializedName("name")
    val name: String? = null,

    @SerializedName("country")
    val country: String? = null,

    @SerializedName("presented_by")
    val presentedBy: String? = null,

    @SerializedName("first_awarded_date")
    val firstAwardedDate: String? = null,

    @SerializedName("image")
    val image: Image,

    @SerializedName("logo")
    val logo: Image,

    @SerializedName("nominees_id")
    val nominees_id: Int? = null,

    @SerializedName("social_list")
    val socialList: List<Social>? = null,
): MainTopic() , Serializable {

    var history: List<Content>? = null
    var nomineesList: List<NomineeResult>? = null
}

File: ios/HistoryCinema/HistoryCinema/Entities/LocalFiles/MainTopics/AwardMainTopic.swift
//
//  AwardMainTopic.swift
//  HistoryCinema
//
//  Created by Tiago Henrique da Silva on 27/02/21.
//

import Foundation
import ObjectMapper

class AwardMainTopic: MainTopic {
    var id: Int!
    var name: String?
    var country: String?
    var presentedBy: String?
    var firstAwardedDate: String?
    var image: LocalImage!
    var logo: LocalImage!
    var nomineesId: Int?
    var socialList: [Social]?
    
    var history: [Content]?
    var nomineesList: [NomineeResult]?
    
    required init?(map: Map) {
        super.init(map: map)
    }

    override func mapping(map: Map) {
        super.mapping(map: map)
        
        id                                                  <- map["id"]
        name                                                <- map["name"]
        country                                             <- map["country"]
        presentedBy                                         <- map["presented_by"]
        firstAwardedDate                                    <- map["first_awarded_date"]
        image                                               <- map["image"]
        logo                                                <- map["logo"]
        nomineesId                                          <- map["nominees_id"]
        socialList                                          <- map["social_list"]
    }
}
